src/pages/api/fetchSummary.ts

```diff
@@ -0,0 +1,62 @@
+import { Configuration, OpenAIApi } from 'openai';
+import { mockData, DataType } from '../../../mock-response';
+import type { NextApiRequest, NextApiResponse } from 'next';
+import { generatePrompt } from '~/utlis/generatePrompt';
+
+const configuration = new Configuration({
+  apiKey: process.env.OPENAI_API_KEY,
+});
+const openai = new OpenAIApi(configuration);
+
+type Error = {
+  error: { message: string };
+};
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<DataType | Error>,
+) {
+  if (!configuration.apiKey) {
+    res.status(500).json({
+      error: {
+        message:
+          'OpenAI API key not configured, please follow instructions in README.md',
+      },
+    });
+    return;
+  }
+
+  const { query } = req;
```

Comment:
Do you think that this request would be better suited as a POST request with a body? I know we're not actually storing any data, but we are hitting ChatGPT with a large amount of text and possibly a JSON object in the future. It might be better to format this to accept the chunkedArray in the body

We could loop through the chunkedArray in one api request rather than hitting this api endpoint multiple times. Also Pratik mentioned we might want to combine our api's together so that the front end only has to request the backend api one time. We can talk about this more tomorrow :)

Reply:
Discussed this on slack and over the call.